core/src/main/java/software/amazon/awssdk/SdkRequest.java

```diff
@@ -13,11 +13,14 @@
  * permissions and limitations under the License.
  */
 
-package software.amazon.awssdk.opensdk.protect.auth;
+package software.amazon.awssdk;
 
-import java.util.Optional;
-import software.amazon.awssdk.auth.RequestSigner;
-
-public interface RequestSignerProvider {
-    Optional<RequestSigner> getSigner(Class<? extends RequestSigner> requestSignerType);
+/**
+ * The base class for all SDK requests.
+ *
+ * TODO: SDK-specific options on the {@link AmazonWebServiceRequest} should be migrated here as part of the base-model refactor.
+ *
+ * @see SdkResponse
+ */
+public class SdkRequest {
```

Comment:
Yes! I'll start with abstract, since all of our models are classes and I don't think we've had problems with AmazonWebServiceRequest not being an interface. Thanks for catching this.